### DISCUSSION TITLE
UML Diagrams?

### DISCUSSION THREAD

**Discussion by VlasanRobert** [JdeRobot/RoboticsAcademy#2388] [Q&A]
Hello 

where can i find some UML diagrams to help me

**Comment by jmplaza** [comment]
Hi @VlasanRobert ,

(I've moved your question to RoboticsAcademy Discussions, it fits better here)

We don't have UML diagrams yet, but let me share here the general design of RoboticsAcademy. It may help you when understanding the source code:

At run time there two actors: (A) the browser and (B) the docker image. The browser shows (a1) the exercise webpage frontend, where the user may write her source code for solving the exercise and see it running on the simulated robot. The frontend is based on REACT library. The exercise webpage is provided by (a2) a web server, based on Django, which is running inside the docker image.

In addition, the docker container allows the dockerized execution of robotics applications and the robotics simulator. It includes (b1) all the robotics dependencies (ROS, Gazebo, OpenCV...) already pre-installed,  all the simulator scenarios and the launcher files. Both are stored in the [RoboticsInfrastructure repository](https://github.com/JdeRobot/RoboticsInfrastructure). We have two branches there, one for ROS1-Noetic and one for ROS2-Humble. 

The docker image also includes (b2) a [RoboticsApplicationManager](https://github.com/JdeRobot/RoboticsApplicationManager) running, which starts the simulator with the world of each exercise, starts the visualization tools and starts the robotics application with the user source code.  The visualization tools show information at the exercise webpage, such as the Gazebo viewer, a console and other specific widgets for debugging.

The robotics application is a combination of user code (who has edited it at the exercise webpage), hal.py (a module providing access to robot sensors and actuators, the HAL, Hardware Abstraction Layer) and gui.py (a Python module providing some visualization methods for debugging).

![RA-design](https://github.com/JdeRobot/RoboticsAcademy/assets/10961573/7a5dc036-f88d-468c-8ff8-af3a79138bd4)

Hope it helps.
